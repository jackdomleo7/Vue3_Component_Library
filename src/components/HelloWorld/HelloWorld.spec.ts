import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { JHelloWorld } from '@/index'

describe.concurrent('HelloWorld', () => {
  describe('props', () => {
    it('displays the message "Hello there!"', () => {
      const wrapper = mount(JHelloWorld, {
        props: {
          msg: 'Hello there!'
        }
      })

      expect(wrapper.find('h1').text()).toBe('Hello there!')
    })
  })

  describe('interactivity', () => {
    it('increments the counter when clicked', async () => {
      const wrapper = mount(JHelloWorld, {
        props: {
          msg: 'Hello there!'
        }
      })

      expect(wrapper.find('button').text()).toBe('count is 0')

      await wrapper.find('button').trigger('click')

      expect(wrapper.find('button').text()).toBe('count is 1')
    })
  })
})
