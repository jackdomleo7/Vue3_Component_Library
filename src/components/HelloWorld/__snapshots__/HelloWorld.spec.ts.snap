// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > snapshots > matches snapshot when mounted 1`] = `
<div
  data-v-app=""
>
  
  <h1
    data-v-1f28467c=""
  >
    Hello there!
  </h1>
  <div
    class="card"
    data-v-1f28467c=""
  >
    <button
      data-v-1f28467c=""
      type="button"
    >
      count is 0
    </button>
    <p
      data-v-1f28467c=""
    >
       Edit 
      <code
        data-v-1f28467c=""
      >
        components/HelloWorld.vue
      </code>
       to test HMR 
    </p>
  </div>
  <p
    data-v-1f28467c=""
  >
     Check out 
    <a
      data-v-1f28467c=""
      href="https://vuejs.org/guide/quick-start.html#local"
      target="_blank"
    >
      create-vue
    </a>
    , the official Vue + Vite starter 
  </p>
  <p
    data-v-1f28467c=""
  >
     Install 
    <a
      data-v-1f28467c=""
      href="https://github.com/johnsoncodehk/volar"
      target="_blank"
    >
      Volar
    </a>
     in your IDE for a better DX 
  </p>
  <p
    class="read-the-docs"
    data-v-1f28467c=""
  >
    Click on the Vite and Vue logos to learn more
  </p>
  
</div>
`;

exports[`HelloWorld > snapshots > matches snapshot when shallow mounted 1`] = `
<div
  data-v-app=""
>
  
  <h1
    data-v-1f28467c=""
  >
    Hello there!
  </h1>
  <div
    class="card"
    data-v-1f28467c=""
  >
    <button
      data-v-1f28467c=""
      type="button"
    >
      count is 0
    </button>
    <p
      data-v-1f28467c=""
    >
       Edit 
      <code
        data-v-1f28467c=""
      >
        components/HelloWorld.vue
      </code>
       to test HMR 
    </p>
  </div>
  <p
    data-v-1f28467c=""
  >
     Check out 
    <a
      data-v-1f28467c=""
      href="https://vuejs.org/guide/quick-start.html#local"
      target="_blank"
    >
      create-vue
    </a>
    , the official Vue + Vite starter 
  </p>
  <p
    data-v-1f28467c=""
  >
     Install 
    <a
      data-v-1f28467c=""
      href="https://github.com/johnsoncodehk/volar"
      target="_blank"
    >
      Volar
    </a>
     in your IDE for a better DX 
  </p>
  <p
    class="read-the-docs"
    data-v-1f28467c=""
  >
    Click on the Vite and Vue logos to learn more
  </p>
  
</div>
`;
